lizmap/modules/lizmap/classes/lizmapWkt.class.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * @author    3liz
+ * @copyright 2020 3liz
+ *
+ * @see      http://3liz.com
+ *
+ * @license Mozilla Public License : http://www.mozilla.org/MPL/
+ */
+class lizmapWkt
+{
+    /** @var regExes[] */
+    protected static $regExes = array(
+            'typeStr' => '/^\s*(\w+)\s*(\w+)?\s*\(\s*(.*)\s*\)\s*$/',
+            'spaces' => '/\s+/',
+            'parenComma' => '/\)\s*,\s*\(/',
+            'doubleParenComma' => '/\)\s*\)\s*,\s*\(\s*\(/',  // can't use {2} here
+            'trimParens' => '/^\s*\(?(.*?)\)?\s*$/'
+    );
+
+    /**
+     * Return a geometry array likes it is defined in GeoJSON.
+     *
+     * @param string $wkt A WKT string
+     *
+     * @return array|null The geometry as described in GeoJSON
+     *
+     */
+    static public function parse($wkt)
+    {
+        // Extracting geometry type, dimension and coordinates
+        preg_match(self::$regExes['typeStr'], $wkt, $matches);
+        if (count($matches) != 4) {
+            return null;
+        }
+        $geomType = strtolower($matches[1]);
+        $dim = strtolower($matches[2]);
+        $str = $matches[3];
+
+        if (substr($geomType, -2) === 'zm') {
+            $geomType = substr($geomType, 0, -2);
+            $dim = 'zm';
+        } else if (substr($geomType, -2) === 'mz') {
+            $geomType = substr($geomType, 0, -2);
+            $dim = 'mz';
+        } else if (substr($geomType, -1) === 'z') {
+            $geomType = substr($geomType, 0, -1);
+            $dim = 'z';
+        } else if (substr($geomType, -1) === 'm') {
+            $geomType = substr($geomType, 0, -1);
+            $dim = 'm';
+        }
+
```

Comment:
`$dim` is not used afterwards. Will lizmap support Z or M or MZ geometries (planed in the future but not yet 3.4 ?)

Reply:
It is like the OpenLayers WKT parser. Supporting Z or M or ZM in Lizmap is not planed yet.